Station.cpp

```diff
@@ -171,48 +174,69 @@ status_t Station::RetrieveStreamUrl() {
     BMallocIO* plsData = HttpUtils::GetAll(fSource, NULL, 100000, &contentType, 2000);
     if (plsData) {
         status = parseUrlReference((const char*)plsData->Buffer(), contentType);
-		if (status != B_OK && contentType.StartsWith(("audio/")))
+		delete plsData;
+		if (status != B_OK && contentType.StartsWith("audio/")) {
 			SetStreamUrl(fSource);
-        delete plsData;
+			return B_OK;
+		}
     }
     return status;
 }
 
 status_t Station::Probe() {
     BHttpHeaders headers;
+    BMallocIO* buffer;
     BUrl* resolvedUrl = new BUrl();
 	BString contentType("audio/*");
 	
 	// If source URL is a playlist, retrieve the actual stream URL
-	if (fSource.Path().EndsWith(".pls") || 
-						fSource.Path().EndsWith(".m3u") || 
-						fSource.Path().EndsWith(".m3u8"))
-                    RetrieveStreamUrl();
+	//if (fSource.Path().EndsWith(".pls") || 
+	//					fSource.Path().EndsWith(".m3u") || 
+	//					fSource.Path().EndsWith(".m3u8"))
+	RetrieveStreamUrl();
 	
-	status_t resolveStatus = HttpUtils::CheckPort(fStreamUrl, resolvedUrl);
-	if (resolveStatus != B_OK)
-		return B_ERROR;
-    BMallocIO* buffer = HttpUtils::GetAll(*resolvedUrl, &headers, 2 * 1000 * 1000, &contentType, 4096);
-    delete resolvedUrl;
+	// FIXME: UGLY HACK!
+	// Currently Haiku's HttpRequest is not able to check for connection on a specified IP address/port pair
+	// before actually resolving the hostname to that pair. That means, if the first server in the DNS
+	// response is down or has its port closed for some reason, the request will simply fail.
+	//
+	// Creating a socket and trying to connect to the specified IP/port pair is enough for our purposes. That
+	// way, if a server is down we can use the next DNS record, and craft a new URL like http://192.168.0.1/path/to/stream
+	// 
+	// On HTTPS, unfortunately, this isn't possible, as the certificate is associated to the hostname, and so we have to
```

Comment:
This one is coming from last year. I'll take a look, I can't remember now but it's possible that I've already opened a bug report. Otherwise I'll just do it.

Reply:
I've filled a [bug report](https://dev.haiku-os.org/ticket/16155#ticket)